### PR TITLE
Use safe DLL loading

### PR DESCRIPTION
**Pull request by kennykerr**
Restricts DLL search directories for safety.

This update corresponds to the following change: https://github.com/microsoft/cppwinrt/pull/1293

### CODE REVIEW

File: crates/libs/windows/src/imp/delay_load.rs [hunk 1]
```diff
@@ -8,7 +8,7 @@ use super::*;
 ///
 /// * Both the library and function names must be valid null-terminated strings.
 pub unsafe fn delay_load<T>(library: core::PCSTR, function: core::PCSTR) -> Option<T> {
-    let library = LoadLibraryA(library);
+    let library = LoadLibraryExA(library, 0, LOAD_LIBRARY_SEARCH_DEFAULT_DIRS);
```
Comment (riverar) [inline crates/libs/windows/src/imp/delay_load.rs:11]: Would it be better to use `LOAD_LIBRARY_SEARCH_SYSTEM32` here? Seems all internal use of delay_load only uses modules from \system32 context.

Comment (kennykerr) [reply crates/libs/windows/src/imp/delay_load.rs:11]: No, both cppwinrt and the windows crate support xcopy activation for WinRT.

**Review by riverar — DISMISSED** [review]

**Comment by kennykerr** [comment]
This test fails because Cargo doesn't provide a reliable way to determine where output binaries are placed.

File: crates/libs/windows/src/imp/delay_load.rs (hunk 1 above)
Comment (Eyryu) [reply crates/libs/windows/src/imp/delay_load.rs:11]: @kennykerr

**Pull request merged**